### PR TITLE
use WaitOnAddress notification

### PR DESCRIPTION
**Pull request by hliu18**
Use notification on address (WaitOnAddress) to wait for completed IOs, get rid of polling with busy loop.

### CODE REVIEW

**Comment by hliu18** [comment]
There may still be build issue. I see lots of warnings not related to this change.

**Comment by harsha-simhadri** [comment]
@hliu18, would be good to get Gopal and Li's review on this as they are more familty with this env.
The changes seem reasonable to me

**Comment by hliu18** [comment]
@litian1 @gopalrs Please help review as well.

**Comment by hliu18** [comment]
fix reviewer.
@ltan1ms @gopalrs can you please review and approve? thanks

File: src/pq_flash_index.cpp [hunk 1]
```diff
@@ -1082,14 +1094,12 @@ namespace diskann {
 #ifdef USE_BING_INFRA
       // process each frontier nhood - compute distances to unvisited nodes
       int completedIndex = -1;
+      long requestCount = (long)frontier_read_reqs.size();
```
Comment (ltan1ms) [inline src/pq_flash_index.cpp:1097]: super nit: `static_cast<long>`

**Review by ltan1ms — APPROVED** [review]

File: src/pq_flash_index.cpp (hunk 1 above)
Comment (hliu18) [reply src/pq_flash_index.cpp]: fixed.

**Comment by harsha-simhadri** [comment]
> There may still be build issue. I see lots of warnings not related to this change.

IS this resolved now?

**Review by gopalrs — APPROVED** [review]

**Comment by hliu18** [comment]
@harsha-simhadri @gopalrs Can one of you also approve so that this PR can be merged? ltan1ms already approved.

**Comment by hliu18** [comment]
@harsha-simhadri the build issue is resolved. Please approve to unblock merge.

**Comment by hliu18** [comment]
I don't have permission to merge. @harsha-simhadri can you help merge?

**Pull request merged**